examples/c: Build bpftool submodule in cmake

JackyYin opened the pull request:
Ref: #70.

Because tools/bpftool no longer exists, we need to build bpftool from git submodule and set the correct executable path in cmake variable `BPFOBJECT_BPFTOOL_EXE`.

JackyYin commented:
BTW, seems like we can remove the `STEP_TARGETS` in `ExternalProject_Add` and this line also: `add_dependencies(${app_stem}_skel libbpf-build)` because libbpf always built before examples even if I remove these lines.

JackyYin commented:
Hello @anakryiko,
Could you give some suggestions about this PR?
I wonder what can I fix to get it merged.
Thanks a lot!

anakryiko commented on examples/c/CMakeLists.txt:
```diff
@@ -25,9 +25,20 @@ ExternalProject_Add(libbpf
   STEP_TARGETS build
 )
 
+ExternalProject_Add(boftool
```
this should probably be `bpftool`, no?

JackyYin replied:
Fixed.
Sorry for the typo.

anakryiko reviewed (changes requested):
I see this follows the approach for building libbpf so looks sane to me. Please fix the typo and can you also add Signed-off-by: with your full name and email just like libbpf/kernel commits have it? Thanks!

JackyYin commented:
> I see this follows the approach for building libbpf so looks sane to me. Please fix the typo and can you also add Signed-off-by: with your full name and email just like libbpf/kernel commits have it? Thanks!

Thanks for your feedback!
I just fixed the typo and added the `Signed-off-by` to my commit message.
The commit is squashed and force-pushed back again.

anakryiko commented:
Thanks!

danobi commented:
lgtm
